docs/src/proposals/epoch_accounts_hash.md

```diff
@@ -0,0 +1,209 @@
+
+If a snapshot is requested to be created *after* the `start slot` but *before*
+the EAH calculation has completed, then it will be impossible to create a
+snapshot with the correct EAH.  The snapshot process will wait until the EAH
+calculation has completed before proceeding.
+
+
+##### 4. `X == stop slot`
+
+The EAH has been calculated for this epoch, and has been included in the `stop
+slot` bank.  No further handling is required; the snapshot does not need to
+contain the EAH.
+
+
+##### 5. `X > stop slot` and `X <= last slot in epoch`
+
+Same as (4).
+
+
+#### Corner Cases
+
+#### Minimum Slots per Epoch
+
+An EAH is requested by `BankForks::set_root()`, which happens while setting
+*roots*.  The EAH is stored into `Bank`s when they are *frozen*.  Banks are
+frozen 32 slots before they are rooted.  For the expected behavior, the EAH
+start slot really should be 32 slots before the stop slot. If the number of
+slots per epoch is small, this can result in surprising behavior.
+
+Example 1: Assume there are 64 slots per epoch.  The EAH start offset is 16
+and the EAH stop offset is 48.  The difference is 32.  So when Bank 48 is
+frozen before Bank 16 is rooted, a new EAH request has not yet been requested;
+the EAH from the previous epoch is still valid and will be used by Bank 48.
+
+Example 2: Assume there are 66 slots per epoch, then the EAH start offset is
+still 16 and the EAH stop offset is now 49.  The difference is now 33.  When
+Bank 49 is frozen, Bank 16 will already have been rooted, and thus sent an EAH
+request; Bank 49 will wait for the new EAH calculation to complete.
+
+Example 3: Assume there are 32 slots per epoch (the minimum allowed).  The EAH
+start offset is 8, and the EAH stop offset is 24.  Similar to Example 1, Bank
+24 is frozen around when Bank 24 *of the previous epoch* is rooted.  This
+ensures that when the EAH is stored, it'll be for the previous epoch.
+
+In these examples the observed behavior of the EAH is different than when using
+the normal 432,000 slots per epoch.  The EAH is still valid and correct with a
+small number of slots per epoch; it now has a delay of one epoch.  Since the
+epochs themselves can be much faster, security is not reduced.
+
+
+#### Warping
+
+Warping introduces corner cases into EAH because many slots may be skipped,
+including the entire range of `start slot` to `stop slot`.
+
+When warping from before `stop slot` to after, the new bank will include the
+existing EAH in its hash during `freeze()`.  If the bank's parent is from
+before `start slot`, then a new EAH calculation will not have been requested.
+This is safe because warping cannot be used on a live cluster; only for a new
+cluster or tests/debugging.  This means _when_ the EAH was calculated is not
+germane.
+
+When warping from before `start slot` to after, an EAH calculation will be
+requested the next time `set_root()` is called.  Therefore the EAH will be
+based on this new bank.  This is also safe and correct.
```

Comment:
Might be good to break this into cases like we do above, the cases being a function of `parent_slot` and `warp_slot`

1. `parent_slot` <= start_slot`, `warp_slot` >= start_slot: Need to compute EAH on next new bank.
2. `parent_slot` > start_slot,  `warp_slot` >= stop_slot: Include pre-existing EAH stored in the snapshot on new bank.

Reply:
Hmm there seems to be some additional edge cases as pointed out here though: https://github.com/solana-labs/solana/pull/27801/files#r1005230517 where EAH is skipped completely. I think the clearest way to outline this is to list all the possible cases/combinations of `parent_slot`, `warp_slot`, `start_slot` and `stop_slot`